### PR TITLE
7903595: Leverage String templates

### PR DESCRIPTION
**Pull request by JornVernee**
Leverage string templates for source code generation. This makes it much easier to see at a glance what the code that is being generated looks like. It's also a very convenient way of writing this kind of code going forward, and is a good test for the new language feature.

Note that I didn't touch the `Constants::emitLayoutField` and `Constants::emitFunctionDescField`, since they do some pretty complex inline generation to render layouts. I didn't feel like this was worth messing with since we are planning to replace the separate constant classes nay way. Though, I did re-write some of the other methods in `Constants` since they were low-hanging fruits.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no review required)

### Issue
 * [CODETOOLS-7903595](https://bugs.openjdk.org/browse/CODETOOLS-7903595): Leverage String templates (**Enhancement** - P4)

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - Committer) ⚠️ Review applies to [8dba2f20](https://git.openjdk.org/jextract/pull/144/files/8dba2f209fbcb619fcf0a055a008466d59859cbc)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jextract.git pull/144/head:pull/144` \
`$ git checkout pull/144`

Update a local copy of the PR: \
`$ git checkout pull/144` \
`$ git pull https://git.openjdk.org/jextract.git pull/144/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 144`

View PR using the GUI difftool: \
`$ git pr show -t 144`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jextract/pull/144.diff">https://git.openjdk.org/jextract/pull/144.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jextract/pull/144#issuecomment-1828376372)

### CODE REVIEW

File: src/main/java/org/openjdk/jextract/impl/Constants.java [hunk 1]
```diff
@@ -432,47 +427,30 @@ private Constant emitFunctionDescField(FunctionDescriptor desc) {
...
-            indent();
-            append(MEMBER_MODS);
-            append(" MemorySegment ");
             NamedConstant segConstant = new NamedConstant(MemorySegment.class);
-            append(segConstant.constantName);
-            append(" = RuntimeHelper.CONSTANT_ALLOCATOR.allocateFrom(\"");
-            append(Utils.quote(Objects.toString(value)));
-            append("\");\n");
-            decrAlign();
+            appendIndentedLines(STR."""
```
Comment (mcimadamore) [inline src/main/java/org/openjdk/jextract/impl/Constants.java:431]: the `emitLayoutString` method could use a similar treatment - at least for some aspects?

**Review by mcimadamore — APPROVED** [review]
Overall looks good - much easier to read!

**Comment by mcimadamore** [comment]
Do we need build changes to enable preview?

File: src/main/java/org/openjdk/jextract/impl/Constants.java (hunk 1 above)
Comment (JornVernee) [reply src/main/java/org/openjdk/jextract/impl/Constants.java:425]: I've sprinkled some more use of interpolation.

**Comment by JornVernee** [comment]
> Do we need build changes to enable preview?

The build already passes `--enable-preview`. I don't think we changed this after the FFM API was finalized.

**Pull request closed without merging**